src/SearchBar.jsx

```diff
@@ -0,0 +1,201 @@
+var React           = require('react')
+	, Popup           = require('./Popup')
+	, activeElement   = require('react/lib/getActiveElement')
+	, List 						= require('./List')
+	, compat          = require('./util/compat')
+	, _               = require('./util/_')
+	, cx 							= require('classnames')
+	, PlainList       = require('./List')
+	, createUncontrolledWidget = require('uncontrollable');
+
+var SearchBar = React.createClass({
+
+	mixins: [
+		require('./mixins/WidgetMixin'),
+		require('./mixins/TimeoutMixin'),
+		require('./mixins/PureRenderMixin'),
+		require('./mixins/DataFilterMixin'),
+		require('./mixins/DataHelpersMixin'),
+		require('./mixins/PopupScrollToMixin'),
+		require('./mixins/RtlParentContextMixin'),
+		require('./mixins/ListMovementMixin')
+	],
+
+	getDefaultProps(){
+		return{
+			className:'',
+			open:false,
+			disabled:false,
+			readOnly:false,
+			busy:false,
+			dropUp:false,
+			messages: msgs()
+		}
+	},
+
+	getInitialState: function(){
+		const data = this.props.data
+			, initialIdx = this._dataIndexOf(data, this.props.value);
+
+
+		return {
+			selected:data[initialIdx],
+			focused:data[initialIdx] || data[0],
+			search:'',
+		}
+	},
+
+	componentWillReceiveProps(props){
+			const data = props.data
+			, idx = this._dataIndexOf(data, props.value);
+		this.setState({
+			search: data[idx],
+			selected: data[idx],
+			focused:  data[!~idx ? 0 : idx]
+		})
+	},
+
+	_data(data = this.props.data, search = this.state.search){
+
+		if(!search.length) return data;
+
+		return data.filter((name)=>{
```

Comment:
`data` should be able to be an array of objects as well strings, you can use the `DataHelperMixin` to handle getting the text values out of a data object, take a look at how the dropdownlist does it.